File: my-folder/0088-merge-sorted-array/solution.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if(n!=0)
        {
            int emptyIndexInNum1=m;
            int startNum1=0;
            int startNum2=0;
            for(int i=startNum2;i<n;i++){
            while(nums1[startNum1]<=nums2[i] && startNum1<emptyIndexInNum1)
               startNum1++;
            for(int k=emptyIndexInNum1;k>startNum1;k--)
            {
                nums1[k]=nums1[k-1];
            }
                emptyIndexInNum1++;
            nums1[startNum1]=nums2[i];
            }
        
        }
        
    }
}

File: my-folder/0088-merge-sorted-array/solution.kt
class Solution {
    fun merge(nums1: IntArray, m: Int, nums2: IntArray, n: Int): Unit {
        var mi= m-1;
        var ni=n-1;
        var ri=m+n-1
        while(mi>=0 && ni>=0)
        {
            if(nums1[mi]>=nums2[ni])
             {
                nums1[ri]=nums1[mi];
                mi--;
             }
             else
             {
                nums1[ri]=nums2[ni];
                ni--;
             }  
             ri--;
        }

        while(ni>=0)
        {
              nums1[ri]=nums2[ni];
                ni--;
                ri--;

        }

        
    }
}
